### ISSUE TITLE
Creating mock meets server 's 500 error

### ISSUE THREAD

**Issue by winkidney** [tomashanacek/mock-server#25]
Hi,
When i use this great tool with tornado4.3(in python 2.6.6 on centos 6.5), I meet an error shown below(with my request).

![_20160121212219](https://cloud.githubusercontent.com/assets/4109722/12481677/aa74fc02-c085-11e5-8a4a-3ea1acfd6990.png)

```
[I 160121 21:26:00 web:1946] 200 POST /__manage/create (192.168.23.1) 3.62ms
[E 160121 21:26:00 web:1524] Uncaught exception GET /__manage (192.168.23.1)
    HTTPServerRequest(protocol='http', host='192.168.23.140:8888', method='GET', uri='/__manage', version='HTTP/1.0', remote_ip='192.168.23.1', headers={'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip, deflate, sdch', 'Host': '192.168.23.140:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8', 'Upgrade-Insecure-Requests': '1', 'Dnt': '1', 'Connection': 'close', 'Referer': 'http://192.168.23.140:8888/__manage/create', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'Cookie': '_xsrf=2|2637eead|1c4ae088fc8d56a6903bb137e8f6a3b0|1453364329; flash_msg_success="2|1:0|10:1453382760|17:flash_msg_success|68:UmVzb3VyY2UgJ3Rlc3RfYXBpJyBoYXMgYmVlbiBzdWNjZXNzZnVsbHkgY3JlYXRlZC4=|c4146521abbb980c079a20ee2ad9288d7b7d8bd8b44dd431bf4d8732edac6e7f"', 'User-Agent': 'Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.111 Safari/537.36'})
    Traceback (most recent call last):
      File "/home/winkidney/.virtualenvs/mock/lib/python2.6/site-packages/tornado/web.py", line 1443, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/home/winkidney/workspace/mock-server/mock_server/handlers.py", line 407, in get
        ("success", "error")))
      File "/home/winkidney/.virtualenvs/mock/lib/python2.6/site-packages/tornado/web.py", line 699, in render
        html = self.render_string(template_name, **kwargs)
      File "/home/winkidney/.virtualenvs/mock/lib/python2.6/site-packages/tornado/web.py", line 806, in render_string
        return t.generate(**namespace)
      File "/home/winkidney/.virtualenvs/mock/lib/python2.6/site-packages/tornado/template.py", line 345, in generate
        return execute()
      File "list_resources_html.generated.py", line 48, in _tt_execute
        if request.headers.has_key('X-Scheme') and request.headers['X-Scheme']:  # list_resources.html:10 (via layout.html:55)
    AttributeError: 'HTTPHeaders' object has no attribute 'has_key'
[E 160121 21:26:00 web:1946] 500 GET /__manage (192.168.23.1) 41.66ms
```

I will add a Pull request for it(safely get the header's value for tornado 4.3).

**Comment by srthorat** [comment]
same with GET


[E 160427 04:42:45 web:1524] Uncaught exception GET /__manage (127.0.0.1)
    HTTPServerRequest(protocol='http', host='localhost:9999', method='GET', uri='/__manage', version='HTTP/1.1', remote_ip='127.0.0.1', headers={'Accept-Language': 'en-US,en;q=0.5', 'Accept-Encoding': 'gzip, deflate', 'Host': 'localhost:9999', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:39.0) Gecko/20100101 Firefox/39.0', 'Connection': 'keep-alive', 'Cookie': '_xsrf=2|5cd8c20a|e28e7d2e1627c7c8c7b2e79e31c31f30|1461756553'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado-4.3-py2.7-linux-x86_64.egg/tornado/web.py", line 1443, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/usr/local/lib/python2.7/dist-packages/mock_server-0.3.7-py2.7.egg/mock_server/handlers.py", line 407, in get
        ("success", "error")))
      File "/usr/local/lib/python2.7/dist-packages/tornado-4.3-py2.7-linux-x86_64.egg/tornado/web.py", line 699, in render
        html = self.render_string(template_name, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/tornado-4.3-py2.7-linux-x86_64.egg/tornado/web.py", line 806, in render_string
        return t.generate(**namespace)
      File "/usr/local/lib/python2.7/dist-packages/tornado-4.3-py2.7-linux-x86_64.egg/tornado/template.py", line 345, in generate
        return execute()
      File "list_resources_html.generated.py", line 48, in _tt_execute
        if request.headers.has_key('X-Scheme') and request.headers['X-Scheme']:  # list_resources.html:10 (via layout.html:55)
    AttributeError: 'HTTPHeaders' object has no attribute 'has_key'
[E 160427 04:42:45 web:1946] 500 GET /__manage (127.0.0.1) 12.77ms

**Comment by winkidney** [comment]
@srthorat Use this fork, or just install tornado version<=4.2 https://github.com/BlinkTunnel/stomach-server

**Comment by srthorat** [comment]
Hi
Thanks worked,

I have requirement like 

App===Sends Post===> Server , server should recive post , extract data and reply with GET  like server===should send GET ====> APP

Can we do it ?

**Comment by winkidney** [comment]
@srthorat Mock-Server does not work in this case.Why just write your web-backend here?
I don't understand what you want.More detail?

**Comment by srthorat** [comment]
Actually we have server which sends post request to phone, in that post it sends recording url , phone should extract recording url and respond back with GET including recording url  and other required details, so recording should streamed to phone by server

**Comment by winkidney** [comment]
@srthorat Send request may not be a good choice.Use **websocket** or jsut **socket** to do what you want.
Most cases, phone has not it's own public ip, it's a problem.
Use websocket or socket to keep a long-term connection with your phone.

BTW: This function should not done in mock server, it's no longer "mock".
May be query string can be parsed in future version:)

**Comment by tomashanacek** [comment]
Should be fixed in version 0.3.8
